src/components/menus/base-equipment-menu.js

```diff
@@ -132,18 +135,17 @@ const BaseEquipmentMenu = ({
                     >
                         {/* menus for the substation */}
                         <ItemViewInSpreadsheet
+                            key={equipment.substationId}
                             equipmentType={equipments.substations}
                             equipmentId={equipment.substationId}
-                            itemText={() => {
-                                const substation = network.getSubstation(
-                                    equipment?.substationId
-                                );
-                                return getNameOrId(substation);
-                            }}
+                            itemText={getNameOrId(
+                                network.getSubstation(equipment.substationId)
```

Comment:
No need of equipment**?**.substationId

Reply:
no we are sure equipement is not null (test at line 130)

Reply:
remark: I changed these lines because we were displaying an empty substation name